concordium-consensus/src/Concordium/Scheduler/Environment.hs

```diff
@@ -989,6 +989,8 @@ data WithDepositContext m = WithDepositContext
       -- | Address of the sender of the transaction.
       -- This should correspond to '_wtcSenderAccount', but need not be the canonical address.
       _wtcSenderAddress :: !AccountAddress,
+      -- | Address of the optional sponsor of the transaction.
+      _wtcSponsorAddress :: !(Maybe AccountAddress),
```

Comment:
I'm not sure if this is needed. At least at the moment, it's not used, so I suggest removing for now.